### DISCUSSION TITLE
Example datasets for training (human and machine) - for CF roadmap

### DISCUSSION THREAD

**Discussion by lhmarsden** [cf-convention/discuss#355] [Comments and ideas for changing CF]
### Topic for discussion

Example datasets would be useful as training datasets for human users. These training datasets could also potentially be used for training a GPT (for example) to 'understand' AI in order to help human users with questions. 

This discussion relates to this theme in the CF roadmap publication that we are collectively working on; the 5 year plan.

**Comment by lhmarsden** [comment]
Could this 'collection' of example datasets be searchable?

**Comment by lhmarsden** [comment]
@agstephens it was great to discuss this with you at the workshop last week. Would be good to hear your thoughts on how we proceed for the purpose of the roadmap publication.

**Comment by sethmcg** [comment]
What are the desirable characteristics of these example datasets?

Are we talking idealized examples where everything is perfect (but perhaps without much real data)?  Or real-world datasets that may have imperfections but are widely used?  Something else?

**Comment by ChrisBarker-NOAA** [comment]
I would say "everything is perfect" examples, that's kind of the point :-)

Many of us learn much more bey example, so if someone can find an example that's similar to their use case, it will be much easier to figure out what to do.

> perhaps without much real data

Yes, I think the actual data isn't the point, so ideally, they'd be trimmed down and small...

**Comment by lhmarsden** [comment]
I think it depends what we are using these datasets for.

For human training, everything should be perfect. It could be that synethetic examples (no real data) could be used for this. Large language models like ChatGPT could help us make these more quickly than we might've been able to do so a few years ago, though this will still take a lot of manual work. Related to this are profiles of CF, e.g. the CF-WMO profiles. These profiles outline explicitely, with fewer degrees of freedom, how to encode certain types of data. Therefore I think there is some overlap with https://github.com/orgs/cf-convention/discussions/358 here.

For machine training, I don't think everything needs to be perfect if we have enough datasets. It could actually be beneficial include labelled imperfections, so the machine could learn what not to do. Though this hypothesis requires a lot of testing...

**Comment by lhmarsden** [comment]
For machine training, in order to gather a large enough data collection (1000+ ???), we may have to use real datasets. We discussed at the workshop that multiple examples from the same collection (e.g. datasets for different days) could be used.

**Comment by sadielbartholomew** [comment]
If the data itself isn't particularly important, and only the metadata, and I think this is generally true, then how about we have our examples in _header-only_ [CDL notation form](https://www.unidata.ucar.edu/software/netcdf/workshops/2012/nc3model/Cdl.html)_ - to keep everything as small as we can i.e. no data arrays included (which are the part that can make files large and therefore difficult to store and transfer), without losing any metadata information? This can be generated from the 'real' datasets quite easily, using `ncdump -h` and then saving the output as `<filename>.cdl>`, so with a small script we can batch convert any real datasets into that format.

**Reply by lhmarsden** [reply]
I second that idea

**Reply by ethanrd** [reply]
I agree in general. Though, I suspect some CF capabilities would benefit from example datasets with some data. Coordinate subsampling comes to mind but I expect there are others.

**Comment by agstephens** [comment]
Hi everyone, apologies for the delay in getting to this thread. I originally proposed this idea at the CF Workshop (2024) so let me share some high-level thoughts based on the discussions we had in the brief hackathon, reading your comments above, and some further thinking. I'll try to break it down as follows:

1. What we tried/learnt at the CF Workshop 2024
2. Some thoughts about how we might progress this work

# 1. What we tried/learnt at the CF Workshop 2024

## 1.1. Large Language Models and CF

We talked about training datasets for a Large Language Model (LLM) such as ChatGPT and some of key points in our discussion were:
- The _language_ that we want an LLM to know is actually CDL (i.e. the output of `ncdump`)
- ChatGPT already has some knowledge and skills around CF and NetCDF
  - It has already seen a lot of open-source material that is relevant (GitHub and documentation)
  - With a few sentences of description (about temperature, times, locations etc), it created a valid CF-compliant CDL header
  - With more prompting it improved the metadata in the header and added some features more specific to recent CF versions
  - Although it made mistakes, further prompting could guide it
- It would be more efficient to investigate simplest solutions first:
  1. Try the free ChatGPT version
  2. Try an [OpenAI Assistant](https://platform.openai.com/docs/assistants/overview) (like an _AI Agent_)
     - Using uploaded files (docs and examples) and a long text prompt (i.e. a set of rules the agent should know)
  3. Try more involved approaches:
     - Retrieval Augmented Generation (RAG)
     - Fine-tuning
- Every few months, it would be wise to retry the simplest solution to check how well it works with the latest publicly available LLMs
- Measuring performance is hard:
   - How do we compare different solutions?
   - How do we measure progress/accuracy?

## 1.2. Creating example datasets

Regarding the creation of a high-quality dataset:
- We noted that synthetic example data might be useful (and quicker to generate)
- There were two distinct use cases:
  1. **Create a small set of files to accompany the conventions documentation:** 
    - To demonstrate implementation of each CF feature
    - To use as best practice examples
    - To use as a template for creating a similar file
    - For use by the CF-Checker
    - Ideally, a unique set of files exists for each CF version
    - 10-20 files might be enough to meet this need
  2. **Create a large set of file headers (CDL) for Machine Learning (ML) applications:**
    - For training LLMs
    - Including data files classified as "high" and "low" quality - so ML classifiers could be training on them
    - For other unknown uses
    - Probably want >10,000 individual files
    - The dataset could be published at: https://paperswithcode.com/datasets
    - Could ChatGPT help us create synthetic examples files?

# 2. Some thoughts about how we might progress this work

Having experimented very briefly with prompting an LLM (ChatGPT) it is clear that there is some promise. There is also a danger that we could put a lot of effort into something that becomes obsolete within months or years. I can think of **6 different areas/approaches** that we could take forward. It would be great to hear which people think is the best combination, and in what order we could tackle them:
- 2.1. Define what we want from a CF-Agent
- 2.2. Create performance metrics and tests
- 2.3. Small file set to support the CF documentation
- 2.4. Large file set to support ML applications
- 2.5. Try a range of ML approaches
- 2.6. Constrain the problem 

## 2.1. Define what we want from a CF-Agent

A real danger is that effort is put into playing with a "CF-Agent" but there is no clear goal, leading to a vague outcome. We might benefit from defining a set of **use cases** and/or **requirements,** such as:
- Able to describe the contents of a file in human language(s)
- Able to check for errors
- Able to advise on best practice
- Able to create CF-compliant files
- Other uses?

## 2.2. Create performance metrics and tests

Any work done with ML models requires us to be able to recognise when the AI is _doing a good job_. We would need to define performance metrics and repeatable tests that could assess the quality of the model/application. The tests would also enable regression testing when updates were made to parts of the system outside of our control.

## 2.3. Small file set to support the CF documentation

This has been mentioned above in [section 1.2](#1.2.-creating-example-datasets). Ideally there would be a set of files to support:
- each version of the CF conventions
- each feature/profile/property mentioned in the conventions document

## 2.4. Large file set to support ML applications

This has been mentioned above in [section 1.2](#1.2.-creating-example-datasets). Some of the key considerations would be:
- crowd-source data files from many organisations and scientific disciplines
- CF-check all files
- Maybe check files can be read by at least three libraries/tools (in multiple languages)
- Use CDL headers only (although _how can we test reading files if we only have the headers_?)
- Aim for >10,000 files
- Need "good" and "bad" files for classification
- How do we identify a _high-quality_ file? Is it more than CF-compliant?
- Publish to `paperswithcode`

##  2.5. Try a range of ML approaches

We should consider a hierarchy of approaches, from the quickest/cheapest first:
1. Free public LLM
2. Agent/Assistant on top of (1)
3. More complex approaches like multi-agent and RAG
4. Fine-tuning of commercial LLM
5. Fine-tuning of open source LLM (such as Llama 3.1)

## 2.6. Constrain the problem 

Many of the above options appear to involve a lot of work. How can we constrain the problem, so that we can try something out quickly? Here are some ideas, please add more:
- Use NetCDF files from a single project (such as CMIP6) as our universe of possible files - use that to test if our proposed approaches will actually work
- Pick an old version of CF (e.g. CF-1.8) and try to train a model to be an expert in only that - we would need less examples, files, rules, etc.

If we find a good solution to these cut-down problems, we can scale them up.

(Apologies this is so long - thanks for staying with it - I look forward to hearing your thoughts and working on it with you :-) )

**Reply by lhmarsden** [reply]
Rukaya Johaadien at GBIF Norway has just developed a program called ChatIPT that might provide some inspiration. It takes unstructured biodiversity data, converts the tables into a Darwin Core Archive and publishes it. I am surprised how well it works. I know Rukaya put a lot of work into this, but a CF-NetCDF creator using an LLM could be useful in some cases.

Video of it in action (7 mins) https://youtu.be/bhAfy06FMpY
You can use it here:
http://68.183.64.100:3000/
Code
https://github.com/gbif-norway/publishgpt

**Reply by lhmarsden** [reply]
It also just word an award! https://www.gbif.org/news/6aw2VFiEHYlqb48w86uKSf/chatipt-system-wins-the-2024-ebbe-nielsen-challenge

**Reply by agstephens** [reply]
@lhmarsden: what is great about this example (ChatIPT) is how focused it is on achieving a clear goal. We can certainly take inspiration from that kind of approach.

**Reply by lhmarsden** [reply]
@agstephens should we ever want to use AI for a converter, the main challenge is working with unstructured input data. 

Perhaps relevant to this, I developed with @filchos a spreadsheet template generator with a configuration for CF-NetCDF. It is designed to help people who prefer to work usually with spreadsheets get their data into some more structured format before they start converting them. The user can select column headers from the full list of CF standard names, and there is a separate sheet for global attributes. I would like to add a separate sheet for variable attributes too at some point.

The template generator is hosted here:
https://www.nordatanet.no/aen/template-generator/config%3DCF-NetCDF

The code is here (anyone reading is welcome to provide feedback using issues or we can work together on developing it further)
https://github.com/SIOS-Svalbard/Nansen_Legacy_template_generator

We published a paper on this over the summer:
https://doi.org/10.5334/dsj-2024-038

**Reply by agstephens** [reply]
Hi @lhmarsden, I just had a quick look through your paper and template generator. Looks good! I agree that this kind of tool could be a very useful part of the process for converting user data/information into NetCDF files. The template generator highlights the complexity in moving from the "scientist" view of the dataset to a correctly and appropriately encoded CF-netCDF view of the dataset. 

I'm intrigued as to how much of the job you might be able to give to a trained AI model. For example, if you provided lots of example text/csv/excel files, then maybe it could read your data and infer the data structures itself, whilst recommending appropriate attributes.

**Reply by lhmarsden** [reply]
Thanks @agstephens 

I work with a lot of scientists who have unstructured data. Sometimes I can't understand the contents of their files (even after a lot of trying). In such cases, I don't think AI will manage either. Even understanding what column headers are can be tricky. Maybe with some back-and-forth dialogue between the user and the AI some progress could be made. I think this is the approach ChatIPT takes.

But of course, there is a sliding scale from structured to unstructured.

**Comment by ChrisBarker-NOAA** [comment]
>  I suspect some CF capabilities would benefit from example datasets with some data.

+1 on this -- I was recently looking at the ragged array format for trajectories, and it was really hard to wrap. my haed around it without and example with data in it.

**Reply by sadielbartholomew** [reply]
Fair enough, I can foresee other use cases too where data is helpful. Though of course if they have data we want it to be small in array terms, so that there is enough to understand the general concept but not more than necessary as to create larger files. Is that reasonable as a principle to aim for, do we think?

**Reply by lhmarsden** [reply]
I think that is reasonable @sadielbartholomew. Enough to be helpful and easily readable. I suppose we could even use your cfplot library to help people visualise the data if the arrays are quite long.

**Comment by Armin-RS** [comment]
There seems to be some overlap with this old issue: https://github.com/cf-convention/cf-conventions/issues/348
Especially with 1.2 i)

**Reply by agstephens** [reply]
> There seems to be some overlap with this old issue: [cf-convention/cf-conventions#348](https://github.com/cf-convention/cf-conventions/issues/348) Especially with 1.2 i)

@Armin-RS, I agree. Maybe it is most useful to move that part of the discussion into [cf-convention/cf-conventions#348](https://github.com/cf-convention/cf-conventions/issues/348), and to focus this current discussion more on the Machine Learning aspect.

**Comment by lhmarsden** [comment]
It has been a while since there has been any activity on this. Does anyone have anything they want to add?

**Reply by agstephens** [reply]
@lhmarsden: Happy New Year!

We have an ML expert team at STFC who recently gave a short course on LLMs. One of the emerging recommendations was that it the commercial LLMs are moving at such a pace that it is unlikely that training your own LLM, or even fine-tuning one, will result in a better outcome than other techniques like RAG and prompt-engineering. 

I wonder if it would be good to schedule a meeting for us to discuss what our latest thinking is and to come up with one or two mini-projects that might help progress our ideas. Who would be up for a chat?

**Reply by lhmarsden** [reply]
Sounds interesting, @agstephens, we can talk about this. I don't have too many meetings scheduled this week or next so you can pick a time. Just avoid the morning of Wednesday 14th.

**Reply by lhmarsden** [reply]
Sounds like there might be some low-hanging fruit at least just providing the CF conventions website and some other things as sources to focus on.

**Reply by agstephens** [reply]
Hi @lhmarsden: how about Tuesday 14th January, 1300 UK time?

**Reply by lhmarsden** [reply]
Perfect!

**Comment by agstephens** [comment]
Hi @lhmarsden @ethanrd @ChrisBarker-NOAA @sethmcg @sadielbartholomew,

On Tuesday 14th January, 1300 UK time: you are welcome to join a chat about this CF discussion topic. Please join here if you are interested:

https://ukri.zoom.us/j/99350987818

**Reply by ethanrd** [reply]
That's a bit early for me but I will try to join. Thanks!

**Reply by sadielbartholomew** [reply]
Thanks for the invite Ag, I will add it to my calendar and should join (unless something urgent comes up).

**Reply by lhmarsden** [reply]
Is anyone joining me?

**Reply by lhmarsden** [reply]
Let's reschedule

**Reply by agstephens** [reply]
So sorry, everyone. I managed to create the Zoom meeting and then didn't put it in my own diary!!! @lhmarsden very sorry.

**Reply by lhmarsden** [reply]
No problem. I can't do this week or next but the one after should be okay. Perhaps later in the day so better for @ethanrd

**Reply by agstephens** [reply]
@lhmarsden: I am mainly free tomorrow. Any chance you have a slot? And again, I'm really sorry about not turning up to my own calendar invite.

**Reply by agstephens** [reply]
@lhmarsden @ethanrd @sadielbartholomew: how about we reschedule for 27, 28, 29 or 30th January, at 4pm UK time?

**Reply by lhmarsden** [reply]
27th we have the CF committee the same day.

That week is actually a bit tricky for me at 4pm. My wife is away a bit so I have to do all the school pick ups.

Monday 3rd Feb or Wednesday 5th Feb okay at 4pm UK time. I am more available earlier in the day.

**Reply by agstephens** [reply]
@lhmarsden: Let's go for Wednesday 5th Feb at 4pm UK time. @ethanrd: would that work for you?

**Reply by sadielbartholomew** [reply]
Sorry all I was also unavailable in the end due to work towards an imminent release set. 5th Feb works or me assuming you want to go ahead if it works for e.g. Ethan too.

**Reply by ethanrd** [reply]
Yes, 5 Feb at 4pm UK time works for me.

**Reply by agstephens** [reply]
Thanks @lhmarsden @ethanrd @sadielbartholomew, I'll create another Zoom meeting - but this time I'll actually put it in my own diary!!! Thanks for your patience.

**Comment by lhmarsden** [comment]
Sorry, I am busy this week. Actually in UK today because my dad is in hospital, so I am not taking on any new meetings and working part time.

Luke
________________________________

**Reply by agstephens** [reply]
Hi @lhmarsden, sorry to hear about your dad. I hope he is okay.

**Comment by agstephens** [comment]
@lhmarsden @sadielbartholomew @ethanrd: here are the Zoom details:

Ag Stephens - STFC UKRI is inviting you to a scheduled Zoom meeting.

Topic: CF Issue chat - Example datasets for training (human and machine) 
Time: Feb 5, 2025 04:00 PM London

Join Zoom Meeting
https://ukri.zoom.us/j/99350987818

Join as companion: https://ukri.zoom.us/j/99350987818?mode=cp

You can test your connection prior to the meeting here:
https://ukri.zoom.us/test

One tap mobile
+442034815240,,99350987818# United Kingdom
+442039017895,,99350987818# United Kingdom

Dial by your location
        +44 203 481 5240 United Kingdom
        +44 203 901 7895 United Kingdom
        +44 208 080 6591 United Kingdom
        +44 208 080 6592 United Kingdom
        +44 330 088 5830 United Kingdom
        +44 131 460 1196 United Kingdom
        +44 203 481 5237 United Kingdom
        0 800 358 2817 United Kingdom Toll-free
        0 800 456 1369 United Kingdom Toll-free
        0 800 031 5717 United Kingdom Toll-free
        0 800 260 5801 United Kingdom Toll-free
Meeting ID: 993 5098 7818
Find your local number: https://ukri.zoom.us/u/aiAX6XqL7

Join by SIP
99350987818@zoomcrc.com

Join by H.323
144.195.19.161 (US West)
206.247.11.121 (US East)
115.114.131.7 (India Mumbai)
115.114.115.7 (India Hyderabad)
159.124.15.191 (Amsterdam Netherlands)
159.124.47.249 (Germany)
159.124.104.213 (Australia Sydney)
159.124.74.212 (Australia Melbourne)
159.124.168.213 (Canada Toronto)
159.124.196.25 (Canada Vancouver)
Meeting ID: 993 5098 7818

**Comment by lhmarsden** [comment]
We could have a subdirectory in the CF conventions repo called `example_datasets` which contains a lot of .cdl files.

If we want to make this searchable, every dataset needs some metadata. One way to do this is to have a sqllite metadata.db file that one can query using SQL. Columns could include:
- filename
- keywords
- featureType

This should be discussed based on what people might want to search through example datasets based on.

I can contribute some time to creating a proof of concept with a few example datasets and a metadata.db file. 

It should be possible to include datasets in the conventions doc dynamically too so I can work on this.

Thoughts before I go ahead and work on something @ethanrd @agstephens @JonathanGregory?

**Reply by agstephens** [reply]
Hi @lhmarsden,

I think the sub-directory for `example_datasets` is a great idea. For overall ease of use, I would propose having both the NetCDF and CDL versions available. Since CDL is text-based, your search mechanism could just be `grep`.

But I understand the desire (and need) for something more structured in terms of metadata. If there is going to be some kind of database, should it also include references to sections of the CF doc that are relevant?

**Comment by ethanrd** [comment]
I also like the idea of an `example_datasets` sub-directory in the conventions repo. I would lean towards the conventions repo CDL collection being restricted to full dataset CDL files for all the example CDL snippets in the conventions document and finding another location for other CDL dataset collections. Perhaps the website repo or maybe a new example dataset (gallery?) repo.

I'm not sure about storing the matching binary netCDF files for each CDL. If we want to do that, we might want to find another platform for that rather than GitHub.

**Reply by lhmarsden** [reply]
I agree, better to keep full examples separate from snippits. 

Regarding also storing netCDF files, I see some advantages of storing both.
- Some people might prefer to have netCDF files to directly compare against their own and to test in their workflows.
- I could imagine someone using them for developing software that understands CF.
- We would have to make sure that all CDL files could be converted to netCDF without errors due to e.g. typos anyway

It is admittedly a bit more work to storing both CDL and netCDF for both files, but might be okay.

**Comment by agstephens** [comment]
@lhmarsden @ethanrd @sadielbartholomew, 

In advance of our meeting on 1st April, I have been playing with what they call **agentic-RAG** in relation to setting up a chatbot with the following features:

- Uses a Large Language Model (LLM) as its main agent (i.e. to respond to chat queries)
- Manages history - to ensure the chatbot doesn't lose the thread of a conversation
- Retrieval Augmented Generation (RAG) - constraining the LLM to respond based on a knowledge base (such as trusted documentation)
- Interaction with tools/functions - allowing the LLM to reason when to use tools to get information (such as querying databases)
- Evaluation framework - *I haven't got to this yet, but it is basically using an LLM and human feedback to assess whether the responses are aligned with the intended purpose.*

I made a short video about it here:

https://drive.google.com/file/d/1AGngfycH2uwfZwo6VBxK4N0HJyPLHH60/view?usp=drive_link

Questions that I would ask about this are:
1. What should our corpus of *trusted material* be? 
2. Which other functions/tools can be give this agent access to, to help it be a real-time CF assistant?
3. What should our main evaluation tests be?
4. How might be we put this in the public sphere for others to test and rate?

**Reply by lhmarsden** [reply]
Great work @agstephens I thought this was really intersting. Looking forward to 1st April!

Would you mind if I shared this video within my group at work? I think it could provide some inspiration to people.

**Reply by agstephens** [reply]
Hi @lhmarsden, feel free to share with anybody who might be interested. I think that the tools (i.e. third-party libraries) are now readily available to really exploit this pattern, so it could potentially be applied and re-used for various applications/domains.

**Comment by agstephens** [comment]
Dear @lhmarsden @ethanrd @sadielbartholomew,

Following our chat last week, I have created a quick GitHub project to put down some tasks and track progress. There are two main issues that I'd really like your thoughts and input on:

- [What should go in the (RAG) Knowledge Base of documents?](https://github.com/users/agstephens/projects/8/views/1?pane=issue&itemId=108552473)
- [Which tools should we wrap for the chatbot to call?](https://github.com/users/agstephens/projects/8/views/1?pane=issue&itemId=108552534)

In both cases I have explained the context, and there is a table at the bottom for you to add more ideas to. 

No pressure, but your well-informed would be greatly received :-)

**Reply by lhmarsden** [reply]
Nice @agstephens .

I am not sure whether previous versions of the conventions would confuse it, as you say - worth testing maybe. But are previous version neccessary if the change log is descriptive enough?

I added a few things but not sure if I have added them to the wrong 'issue'!

**Reply by agstephens** [reply]
@lhmarsden, thanks for getting started on this.

I think we probably need to experiment with including multiple versions of the CF standard, and see how the chatbot responds.

Thanks for your responses. I think you are right, those that refer to _structured data_, such as controlled vocabularies, are better placed in the other issue (https://github.com/users/agstephens/projects/8/views/1?pane=issue&itemId=108552534) - please could you move them into there. :-)

**Comment by efisher008** [comment]
Hi Ag @agstephens and others,

I am glad there seems to have been good progress made on this so far following the 2024 workshop. Would you be interested in proposing a hackathon on the topic, for the upcoming [2025 workshop on 22-25 September](https://github.com/orgs/cf-convention/discussions/415)? There is an active discussion thread for these [hackathon calls](https://github.com/orgs/cf-convention/discussions/422), and it would be great to capture it here if you would like a focused session to gather interested participants to work on this. 

Best wishes,
Ellie

**Reply by agstephens** [reply]
Hi @efisher008,

Thanks for your message. Yes, it would be good to propose a Hackathon session. I'll have a think about how we might pose it so that we can make the best use of expert time. 

Thanks,

Ag

**Comment by davidhassell** [comment]
Hello,

The second half (24-25 September) of the [2025 CF workshop](https://cfconventions.org/Meetings/2025-Workshop.html) is dedicated to the CF roadmap, including hackathons dedicated to advancing the road map topics. These hackathons have no pre-determined structure and can be used for anything from further debate to writing an actual proposal - whatever would be most useful the topic at this time.

If there is interest in working on this roadmap topic in this way, please let us know in a comment on this discussion soon, so that we can advertise it on the meeting agenda - this is a good opportunity to possibly bring in fresh ideas from people who may not yet have engaged with the roadmap.

Thanks, on behalf of the workshop committee.

**Comment by agstephens** [comment]
Hi David,

Thanks for the prompt.

I am doing my MSc project on trying to build a trustworthy AI agent to help with the CF conventions. Around the time of the CF conference I will need to have something substantial to show and tell. I will also be working on a dataset of queries and responses that can be used to evaluate such an agent.

It would be great to run some kind of session/hackathon on this topic. Maybe it could be split between:

  *
A quick overview of the tool and the process behind it
  *
Allow people to actually use the tool
  *
Gather feedback on the performance and responses experienced when using the tool

That is an initial first idea, I invite all recipients to include your thoughts on how we could make the most productive of time and expertise.

Thanks,

Ag
________________________________

**Comment by lhmarsden** [comment]
I will try to summarise what was discussed in the hackathon at the workshop yesterday. @agstephens @ChrisBarker-NOAA @sadielbartholomew Rob Jacob, Tomasz Kopec, please add any additions/corrections if required.

## Very quick summary for those who don't want to read further
- A new repository has been created for development work on this. Future discussion on this topic should be raised as issues/discussions in this repository, not here. 
https://github.com/cf-convention/cf-example-datasets 
- I suggest we **keep this discussion open but focused only on the content of the roadmap paper**. At least until we have a dedicated place for discussing/drafting that.

## A longer summary of the hackathon discussion and progress

### Goal

To develop a catalogue of example datasets that:
- are fully CF compliant
- exist in .nc and .cdl format
- to begin with, these should only include CF (no attributes from other conventions, no custom terms, as this could confuse uses into thinking these are required for CF compliance)
- are accompanied by a sidecar metadata file (in e.g. JSON, YAML format), which could be used to build a searchable index
- include a small amount of data
- are built using code that will also be stored within the repository and may be be useful to some users who want to replicate creating the example datasets
- can be found through a searchable catalogue based on certain metadata terms (e.g. featureType, keywords, TBD).
- are updated through time where necessary as new best practices emerge in CF.

### Possible foreseen use cases

- A searchable catalogue for the website.
- Some examples in the conventions document have a full example to accompany them (link to full example button)
- All example data files are used in the CF Checker unit tests
- On request from users: we may be able to create versions of some files for compatibility with previous CF releases
- This may be useful for the LLM CF chatbot @agstephens is working on and other tools

### Where might example datasets be sourced from?

We decided that example datasets could be taken from a variety of sources
- Expanding already almost complete examples from the conventions documentation
- Test files already used in software such as cf-python, any compliance checker etc.
- Good publicly available datasets

In all cases, the above should be 'cleaned' to an example dataset that includes only CF.

### Progress
The repository has been created and a folder structure established.

An initial milestone has been added, with a handful of minor issues:

https://github.com/cf-convention/cf-example-datasets/milestone/1
